application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java

```diff
@@ -182,4 +186,17 @@ public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
         throw new UnsupportedOperationException("Not used");
     }
 
+    private void deletePinnedAnnouncement(ThreadChannel channel) {
+        channel.getHistoryFromBeginning(FIRST_TEN)
```

Comment:
@Tais993 better idea how to approach it? if not, id say its okay

Reply:
Why not just listen to all messages, and then remove the "... pinned a message to this message, see all pinned messages". if it happens in that specific channel.

Regardless, this is some weird code, it's intentionally delaying everything with a couple of seconds.
Maybe there is some specific motivation behind that, but it looks very weird to delay 1 queue with 5s, the other with 10s.

Reply:
agreed. just making a message listener that deletes any "pinned" message in all threads in `#questions` is likely easier and cleaner

Reply:
very well, i'll get started on this.